sdks/unity/AgonesSdk.cs

```diff
@@ -325,6 +345,29 @@ private void OnRequestCompleted(AsyncOperation _)
                 continuation = null;
             }
         }
+
+        /// <summary>
+        /// Custom UnityWebRequest http data handler
+        /// that fires a callback whenever it receives data
+        /// from the SDK.Watch() REST endpoint 
+        /// </summary>
+        private class GameServerHandler : DownloadHandlerScript
+        {
+            private WatchGameServerCallback callback;
+            public GameServerHandler(WatchGameServerCallback callback)
+            {
+                this.callback = callback;
+            }
+
+            protected override bool ReceiveData(byte[] data, int dataLength)
+            {
+                string json = Encoding.UTF8.GetString(data);
+                var dictionary = (Dictionary<string, object>) Json.Deserialize(json);
+                var gameServer = new GameServer(dictionary["result"] as Dictionary<string, object>);

```

Comment:
Check to verify "result" key exists in the dictionary?

Reply:
If it doesn't, you will end up with an empty non-populated GameServer.

Should we throw an Exception if it didn't work as expected?

Reply:
Hmnn. Actually I don't mind if the exception occurs. It's probably a better indicator that something went horribly wrong, rather than giving a non-populated GS.

So I'm going to go with "works as intended" :+1: (unless you have an objection?)

Appreciate the explanations!

Reply:
The exception is hard to debug without more details. However, if it is intended, LGTM.